ground/src/main/java/com/google/android/ground/persistence/local/LocalDatabaseModule.kt

```diff
@@ -18,22 +18,30 @@ package com.google.android.ground.persistence.local
 import android.content.Context
 import androidx.room.Room
 import com.google.android.ground.Config
+import com.google.android.ground.coroutines.IoDispatcher
 import com.google.android.ground.persistence.local.room.LocalDatabase
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
 import dagger.hilt.android.qualifiers.ApplicationContext
 import dagger.hilt.components.SingletonComponent
 import javax.inject.Singleton
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.asExecutor
 
 @InstallIn(SingletonComponent::class)
 @Module
 object LocalDatabaseModule {
   @Provides
   @Singleton
-  fun localDatabase(@ApplicationContext context: Context): LocalDatabase {
+  fun localDatabase(
+    @ApplicationContext context: Context,
+    @IoDispatcher ioDispatcher: CoroutineDispatcher
+  ): LocalDatabase {
     return Room.databaseBuilder(context, LocalDatabase::class.java, Config.DB_NAME)
       .fallbackToDestructiveMigration() // TODO(#128): Disable before official release.
+      // Run queries and transactions on background I/O thread.
+      .setQueryExecutor(ioDispatcher.asExecutor())
       .build()
   }
```

Comment:
nit: this could be a single-expression function

Reply:
Fixed, ty.